src/torchpme/tuning/tuner.py

```diff
@@ -0,0 +1,325 @@
+    If the error is smaller than the accuracy, the timing is measured and returned.
+    If the error is larger than the accuracy, the timing is set to infinity and the
+    parameter is skipped.
+
+    :param charges: atomic charges
+    :param cell: single tensor of shape (3, 3), describing the bounding
+    :param positions: single tensor of shape (``len(charges), 3``) containing the
+        Cartesian positions of all point charges in the system.
+    :param cutoff: real space cutoff, serves as a hyperparameter here.
+    :param calculator: the calculator to be tuned
+    :param params: list of Fourier space parameter sets for which the error is estimated
+    :param exponent: exponent of the potential, only exponent = 1 is supported
+    :param neighbor_indices: torch.Tensor with the ``i,j`` indices of neighbors for
+        which the potential should be computed in real space.
+    :param neighbor_distances: torch.Tensor with the pair distances of the neighbors
+        for which the potential should be computed in real space.
+    """
+
+    def __init__(
+        self,
+        charges: torch.Tensor,
+        cell: torch.Tensor,
+        positions: torch.Tensor,
+        cutoff: float,
+        calculator: type[Calculator],
+        error_bounds: type[TuningErrorBounds],
+        params: list[dict],
+        exponent: int = 1,
+        neighbor_indices: Optional[torch.Tensor] = None,
+        neighbor_distances: Optional[torch.Tensor] = None,
+    ):
+        super().__init__(
+            charges,
+            cell,
+            positions,
+            cutoff,
+            calculator,
+            exponent,
+        )
+        self.error_bounds = error_bounds
+        self.params = params
+        self.time_func = TuningTimings(
+            charges,
+            cell,
+            positions,
+            cutoff,
+            neighbor_indices,
+            neighbor_distances,
+            True,
+        )
+
+    def tune(self, accuracy: float = 1e-3) -> tuple[list[float], list[float]]:
+        """
+        Estimate the error and timing for each parameter set. Only parameters for
+        which the error is smaller than the accuracy are timed, the others' timing is
+        set to infinity.
+
+        :param accuracy: a float, the desired accuracy
+        :return: a list of errors and a list of timings
+        """
+        if not isinstance(accuracy, float):
+            raise ValueError(f"'{accuracy}' is not a float.")
+        smearing = self.estimate_smearing(accuracy)
+        param_errors = []
+        param_timings = []
+        for param in self.params:
+            error = self.error_bounds(smearing=smearing, cutoff=self.cutoff, **param)  #  type: ignore[call-arg]
+            param_errors.append(float(error))
+            if error > accuracy:
+                param_timings.append(float("inf"))
+                continue
+
+            param_timings.append(self._timing(smearing, param))
```

Comment:
Maybe avoidinb the continue?

```suggestion
            if error > accuracy:
               timing = float("inf")
            else:
                timing = self._timing(smearing, param)

            param_timings.append(timing)
```